core/src/main/java/io/kestra/core/tasks/executions/Labels.java

```diff
@@ -0,0 +1,146 @@
+import io.kestra.core.serializers.JacksonMapper;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+import org.slf4j.Logger;
+
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.kestra.core.utils.Rethrow.throwBiConsumer;
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Allow to add labels to current execution at runtime.",
+    description = "Can be used to ease filtering executions by some runtime-fetched data."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Add labels based on a webhook payload",
+            full = true,
+            code = {
+                "id: webhook-based-labels",
+                "namespace: com.company",
+                "tasks:",
+                "  - id: update-labels-with-map",
+                "    type: io.kestra.core.tasks.executions.Labels",
+                "    labels:",
+                "      customerId: \"{{trigger.body.customerId}}\"",
+                "  - id: by-list",
+                "    type: io.kestra.core.tasks.executions.Labels",
+                "    labels:",
+                "      - key: orderId",
+                "        value: \"{{trigger.body.orderId}}\"",
+                "      - key: orderType",
+                "        value: \"{{trigger.body.orderType}}\"",
+                "triggers:",
+                "  - id: webhook",
+                "    key: order-webhook",
+                "    type: io.kestra.core.models.triggers.types.Webhook",
+                "    conditions:",
+                "      - type: io.kestra.core.models.conditions.types.VariableCondition",
+                "        expression: \"{{ trigger.body.customerId is defined and trigger.body.orderId is defined and trigger.body.orderType is defined }}\""
+            }
+        )
+    }
+)
+public class Labels extends Task implements ExecutionUpdatingTask {
+    @Schema(
+        title = "Labels to add to current execution.",
+        description = "The value should result in a list of labels or a labelKey:labelValue map",
+        anyOf = {
+            String.class,
+            List.class,
+            Map.class
+        }
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private Object labels;
+
+    @Override
+    public Execution update(Execution execution, RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        Object labelsObject = labels;
+        if (labels instanceof String labelStr) {
+            try {
+                labelsObject = JacksonMapper.toObject(runContext.render(labelStr));
```

Comment:
Please define the mapper as a constant as we usually do

Reply:
That's already a constant behind it, I'm using a static method :+1: